core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/IssueFilters.java

```diff
@@ -28,21 +29,39 @@
 @BatchSide
 public class IssueFilters {
   private final IssueFilter[] filters;
+  private final org.sonar.api.issue.batch.IssueFilter[] deprecatedFilters;
   private final Project project;
 
-  public IssueFilters(Project project, IssueFilter[] exclusionFilters) {
+  public IssueFilters(Project project, IssueFilter[] exclusionFilters, org.sonar.api.issue.batch.IssueFilter[] filters) {
     this.project = project;
     this.filters = exclusionFilters;
+    this.deprecatedFilters = filters;
+  }
+
+  public IssueFilters(Project project, IssueFilter[] filters) {
+    this(project, filters, new org.sonar.api.issue.batch.IssueFilter[0]);
+  }
+
+  public IssueFilters(Project project, org.sonar.api.issue.batch.IssueFilter[] deprecatedFilters) {
+    this(project, new IssueFilter[0], deprecatedFilters);
   }
 
   public IssueFilters(Project project) {
-    this(project, new IssueFilter[0]);
+    this(project, new IssueFilter[0], new org.sonar.api.issue.batch.IssueFilter[0]);
   }
 
-  public boolean accept(String componentKey, org.sonar.api.batch.sensor.issue.Issue rawIssue) {
+  public boolean accept(String componentKey, DefaultClientIssue rawIssue) {
     IssueFilterChain filterChain = new DefaultIssueFilterChain(filters);
     FilterableIssue fIssue = new DefaultFilterableIssue(project, rawIssue, componentKey);
-    return filterChain.accept(fIssue);
+    if (filterChain.accept(fIssue)) {
+      return acceptDeprecated(componentKey, rawIssue);
+    }
+
+    return false;
   }
 
+  public boolean acceptDeprecated(String componentKey, DefaultClientIssue rawIssue) {
+    Issue issue = new DeprecatedIssueAdapterForFilter(project, rawIssue, componentKey);
+    return new DeprecatedIssueFilterChain(deprecatedFilters).accept(issue);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "DeprecatedIssueFilterChain"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)